update.ps1

```diff
@@ -1,201 +1,532 @@
+    
+    #region Create access token
+    $actionMessage = "creating access token"
+    
+    $createAccessTokenBody = @{
+        grant_type    = "client_credentials"
+        client_id     = $actionContext.Configuration.EntraIDAppId
+        client_secret = $actionContext.Configuration.EntraIDAppSecret
+        scope         = $actionContext.Configuration.KPNMWPScope
+    }
+    
+    $createAccessTokenSplatParams = @{
+        Uri             = "https://login.microsoftonline.com/$($actionContext.Configuration.EntraIDTenantID)/oauth2/v2.0/token/"
+        Headers         = $headers
+        Method          = "POST"
+        ContentType     = "application/x-www-form-urlencoded"
+        UseBasicParsing = $true
+        Body            = $createAccessTokenBody
+        Verbose         = $false
+        ErrorAction     = "Stop"
+    }
+    
+    $createAccessTokenResonse = Invoke-RestMethod @createAccessTokenSplatParams
+    
+    Write-Verbose "Created access token. Result: $($createAccessTokenResonse | ConvertTo-Json)"
+    #endregion Create access token
+    
+    #region Create headers
+    $actionMessage = "creating headers"
+    
+    $headers = @{
+        "Authorization"   = "Bearer $($createAccessTokenResonse.access_token)"
+        "Accept"          = "application/json"
+        "Content-Type"    = "application/json;charset=utf-8"
+        "Mwp-Api-Version" = "1.0"
+    }
+    
+    Write-Verbose "Created headers. Result: $($headers | ConvertTo-Json)."
+    #endregion Create headers
+
+    #region Get account
+    # API docs: https://mwpapi.kpnwerkplek.com/index.html, specific API call: GET /api/users/{identifier}
+    $actionMessage = "querying account where [$($correlationField)] = [$($correlationValue)]"
+
+    $getKPNLisaAccountSplatParams = @{
+        Uri         = "$($actionContext.Configuration.MWPApiBaseUrl)/users/$($actionContext.References.Account)"
+        Method      = "GET"
+        Body        = @{
+            select = "$($accountPropertiesToQuery -join ',')"
+        }
+        Verbose     = $false
+        ErrorAction = "Stop"
     }
-    $PreviousPerson = Invoke-RestMethod @LisaRequest @SplatParams
 
-    $outputContext.PreviousData = $PreviousPerson | Select-Object -Property ([array] $outputContext.Data.PSObject.Properties.Name)
+    Write-Verbose "SplatParams: $($getKPNLisaAccountSplatParams | ConvertTo-Json)"
+
+    # Add header after printing splat
+    $getKPNLisaAccountSplatParams['Headers'] = $headers
+
+    $getKPNLisaAccountResponse = $null
+    $getKPNLisaAccountResponse = Invoke-RestMethod @getKPNLisaAccountSplatParams
+    $correlatedAccount = $getKPNLisaAccountResponse
+        
+    Write-Verbose "Queried account where [$($correlationField)] = [$($correlationValue)]. Result: $($correlatedAccount | ConvertTo-Json)"
+    #endregion Get account
+
+    #region Calulate action
+    $actionMessage = "calculating action"
+    if (($correlatedAccount | Measure-Object).count -eq 1) {
```

Comment:
Omdat we de bij de 'GET" op regel 181 op _identifier'_ een entiteit uit de API ophalen, kunnen we nooit meerdere objecten terugkomen. De _identifier_ is een unieke waarde binnen KPN Lisa. Dus, het object bestaat (dan is `$correlatedAccount gevuld) of; het account bestaat niet en resulteert in een _404 NotFound_. In dat geval ga je direct naar de 'catch' op regel 506.

Reply:
In theorie, ja, maar dat betekent dus ook dat deze check verder geen kwaad kan.